utils/ccdatofhir/mapping_datatypes.bal

```diff
@@ -0,0 +1,218 @@
+        }
+        return name;
+    }
+    return;
+}
+
+public isolated function mapCcdaCodingtoFhirCodeableConcept(xml<xml:Element> codingElement) returns r4:CodeableConcept|error {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        r4:CodeableConcept codeableConcept = {};
+        r4:Coding mapCCDACodingtoFHIRCodeResult = check mapCcdaCodingtoFhirCode(codingElement);
+        codeableConcept.coding = [mapCCDACodingtoFHIRCodeResult];
+        return codeableConcept;
+    }
+    return {};
+}
+
+public isolated function mapCcdaCodingtoFhirCode(xml<xml:Element> codingElement) returns r4:Coding|error {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        r4:Coding coding = {};
+        string|error codeVal = codingElement.code;
+        if codeVal is string {
+            coding.code = codeVal;
+        }
+        string|error systemVal = codingElement.codeSystem;
+        if systemVal is string {
+            coding.system = systemVal;
+        }
+        return coding;
+    }
+    return {};
+}
+
+public isolated function mapCcdaGenderCodetoFhirGender(xml<xml:Element> codingElement) returns r4:PatientGender|error? {
+    if codingElement is xml:Element && codingElement.length() > 0 {
+        string|error codeVal = codingElement.code;
+        if codeVal is string {
+            match codeVal {
+                "M" => {
+                    return r4:CODE_GENDER_MALE;
+                }
+                "F" => {
+                    return r4:CODE_GENDER_FEMALE;
+                }
+                "UN" => {
+                    return r4:CODE_GENDER_OTHER;
+                }
+                _ => {
+                    return r4:CODE_GENDER_UNKNOWN;
+                }
+            }
+        }
+    }
+    return;
+}
+
+public isolated function mapCcdaValueToFhirCategory(xml:Text codingElement) returns r4:AllergyIntoleranceCategory|error? {
+    string|error codeVal = codingElement.data();
+    if codeVal is string {
+        match codeVal {
+            "416098002" => {
+                return r4:CODE_CATEGORY_FOOD;
+            }
+            "59037007" => {
+                return r4:CODE_CATEGORY_MEDICATION;
+            }
+        }
+    }
+    return;
+}
+
+public isolated function mapCcdaDateTimeToFhirDateTime(xml<xml:Element> dateTimeElement) returns r4:dateTime|error {
+    if dateTimeElement is xml:Element && dateTimeElement.length() > 0 {
+        string|error dateTimeVal = dateTimeElement.value;
+        if dateTimeVal is string {
+            int length = dateTimeVal.length();
+            if length == 4 {
+                return dateTimeVal;
+            } else if length == 6 {
+                dateTimeVal = dateTimeVal.substring(0, 4) + "-" + dateTimeVal.substring(4, 6);
+            } else if length == 8 {
+                dateTimeVal = dateTimeVal.substring(0, 4) + "-" + dateTimeVal.substring(4, 6) + "-" + dateTimeVal.substring(6, 8);
```

Comment:
Use string template expression. Refer: https://ballerina.io/learn/by-example/backtick-templates/

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.r4/commit/bd15548be31a7e76b4962973508cdedd6f8fd58a